dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ArgumentTypesResolver.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations.processor.data;
+
+import com.google.auto.common.MoreElements;
+import com.google.auto.common.Visibility;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.conjure.java.lib.SafeLong;
+import com.palantir.dialogue.RequestBody;
+import com.palantir.dialogue.annotations.processor.ArgumentType;
+import com.palantir.dialogue.annotations.processor.ArgumentType.OptionalType;
+import com.palantir.dialogue.annotations.processor.ArgumentTypes;
+import com.palantir.dialogue.annotations.processor.ErrorContext;
+import com.palantir.dialogue.annotations.processor.ImmutableOptionalType;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.ri.ResourceIdentifier;
+import com.palantir.tokens.auth.BearerToken;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.TypeName;
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+public final class ArgumentTypesResolver {
+
+    private static final String VALUE_OF_METHOD_NAME = "valueOf";
+
+    /**
+     * Why not generate this by inspecting the {@link com.palantir.dialogue.PlainSerDe} interface?
+     * If we suddenly decide to provide special serialization for a widely used type, we could break wire
+     * compatibility.
+     */
+    private static final ImmutableMap<TypeName, String> PLAIN_SER_DE_TYPES =
```

Comment:
as a FLUP I'll switch this to dialogue-annotations specific interface, that handles more types, so we're not tied to the conjure one.